phoenix-core/src/main/java/org/apache/phoenix/expression/function/FloorWeekExpression.java

```diff
@@ -36,8 +37,7 @@ public FloorWeekExpression(List<Expression> children) {
     }
 
     @Override
-    public long roundDateTime(DateTime datetime) {
-        return datetime.weekOfWeekyear().roundFloorCopy().getMillis();
+    public long roundDateTime(Date dateTime) {
+        return dateTime.getTime() - dateTime.getTime() % week - correction;
```

Comment:
This can't be done using the floor function in DateTimeUtils and Calendar.WEEK_OF_YEAR? 

As a side note, I see in the docs in https://phoenix.apache.org/language/functions.html#floor that we only support DAY, HOUR, MINUTE, SECOND and MILLISECOND, so I'm surprised this was here...

Reply:
I have to check this, to be honest I've done this patch more than a year ago but we because of 4.x branch we delayed it until that is dropped and we can use java8.

https://phoenix.apache.org/language/functions.html#floor page is generated with an old phoenix, that is not really up to date as far as I know.